### PR TITLE
Change to target based CMake build and add build exports

### PR DESCRIPTION
**Pull request by aradi**
- Library is built internally using targets

- Export file for CMake based build (with proper targets) added

- Export file for pkg-conf based build added

- Tests for CMake and pkg-conf based integration added

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,39 +1,57 @@
...
+#
+install(EXPORT mpifx-targets
+  FILE mpifx-targets.cmake
+  NAMESPACE MpiFx::
+  DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/mpifx")
+
+include(CMakePackageConfigHelpers)
+configure_package_config_file(
+  ${CMAKE_SOURCE_DIR}/utils/cmake/mpifx-config.cmake.in
+  ${CMAKE_BINARY_DIR}/cmake/mpifx-config.cmake
```
Comment (awvwgk) [inline CMakeLists.txt:38]: Any reason for not using `CMAKE_CURRENT_SOURCE_DIR` and `CMAKE_CURRENT_BINARY_DIR` here? `CMAKE_SOURCE_DIR` would point to the source directory of the root project, which is not necessarily `mpifx` when used in a subproject, or am I mistaken?

Comment (aradi) [reply CMakeLists.txt]: Absolutely correct, thx!

**Review by awvwgk — COMMENTED** [review]
Some comments so far regarding this build system changes.

File: README.rst [hunk 2]
```diff
@@ -1,55 +1,94 @@
 ****************************************
-MPIFX - Modern Fortran Interface for MPI
+MpiFx - Modern Fortran Interface for MPI
```
[the remark this answers is not in the document]
Comment (aradi) [reply README.rst:2]: Also independent of CMake's convention, I'd prefer not to use uppercased project names. (FORTRAN vs Fortran :wink:) And if we change it anyway, we can just use the same name as project is called in CMake. ("Fx" actually stands for "fix", so the case convention would be even logical.)

**Pull request merged**